### ISSUE TITLE
Break all processes when one process breaks (mult-target debugging)

### ISSUE THREAD

**Issue by jwberck** [microsoft/vscode#137953]
Currently with multi-target debugging, multiple processes can be started and debugged simultaneously. However, when one program breaks, the other one will keep running until manually stopped. Visual Studio has the capability to stop all processes when one process breaks, bringing this to VS Code would open up a lot of capability that doesn't currently exist.

**Comment by weinand** [comment]
@jwberck with "stop all processes when one process breaks" you mean that the other programs are paused (suspended) in some random location?

**Comment by jwberck** [comment]
> @jwberck with "stop all processes when one process breaks" you mean that the other programs are paused (suspended) in some random location?

Correct, let me illustrate how I could see this done in GDBs case. If you have a primary process that you hit a breakpoint on, you could automatically send a "break" command to a secondary process which would set a breakpoint at the next line of execution for that secondary process. Both processes would then be suspended at a breakpoint and ready for inspection via the typical VS Code interface. I'll elaborate on how this could maybe be done generically through VS Code at the end of this comment.

This could be generally useful in client server architectures, but for my specific case I'm running avionics flight software and simulations. The simulation can run away very quickly without an automatic break since the dynamics change so rapidly. Being able to inspect the full simulation state at the point of breaking in the FSW would be brand new capability. This would also allow us to potentially continue using the same flight software and sim processes after the break. Simulations run away so quickly that we would typically need to restart the flight software and simulator to get them synced up again, this feature could potentially reduce the need of that. 

So there are multiple potential benefits for those of us in the flight software and simulation space, but obviously this would help anyone dealing with two processes they want to be able to check when one breaks. This exists within Visual Studio, so it can be done, and it seems like it could be done on VS Code as well.

Now I'll illustrate how I could see this being done, not knowing much about the VS Code guts, but I'll take a shot based on how the GDB server works. The GDB server, which serves as the backend of the VS Code debugging interface for C/C++, can take commands to set breaks on lines and report out when it has broken on a line. There is surely a system that already exists that manages these interactions to produce the GUI frontend for VS Code. If a config flag was added for "compound" launch types, say "breakAllProcessesOnOneBreak", we would simply check that flag and then run a function if it is true "breakAllProcesses". VS Code could define some general break command that is then sent to the generic processes handlers. The extensions handling the specific backend API calls for those processes (GDB and C/C++ extension in this case) would then send a "break" command to those processes, thus breaking the program on the next line and completing the feature.

**Comment by andrewbgross** [comment]
I would like to echo @jwberck's request.  

I've inherited a project that uses Celery to create, reuse, and destroy "worker" processes.  Due to poor design, which I am hopeful we will eventually change, there's actually quite a lot of dynamic data that is shared between the processes-- the application actually isn't a good candidate for a multiprocess model, but using multiple threads isn't (or at least wasn't at the time) an option because it's all written in Python.

In any event, my job would literally be ten-times easier if I could set a breakpoint, and when one of the worker processes hits that breakpoint, all the processes pause/break.   Stepping through any code at that point becomes a dicey proposition, but I don't even need to be able to do that for this to be a Killer feature for me.  Just being able to look at the call stack and local variables for the process that hit the breakpoint without all the other processes changing values in the shared Redis cache would be a huge win; being able to look at the call stacks and local variables for any process that seems to be related to the issue I'm debugging would be life-changing.  While I can imagine that even a slight discrepancy in the time at which each process gets paused would make the data so inconsistent as to be useless for some applications, the one I'm debugging does a lot of database reads/writes and sending/receiving regular HTTP traffic, so for my purposes, I imagine that even a full second or two delay in some processes being paused after one of the processes hits a breakpoint would still leave the data consistent enough for my purposes 99% of the time.
